client/state/plugins/wporg/reducer.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import React from 'react/addons';
+
+function updatePluginState( state = {}, pluginSlug, attributes ) {
+	return Object.assign( {},
+		state,
+		{ [ pluginSlug ]: Object.assign( {}, state[ pluginSlug ], attributes ) }
+	);
+}
```

Comment:
well, this would be easy to change to use react.addons.update... but I'm all for `Object.assign` and keep `React.addons.update` just for when it's needed because how the data is structured. 

Anyway, if most of us prefer to use that react.addons.update for consistency, my arguments for object.assign are not so strong that I would want to break the standard :)